korrekt/src/circuit_analyzer/analyzer.rs

```diff
@@ -776,6 +918,170 @@ impl<'b, F: AnalyzableField> Analyzer<F> {
         }
         Ok(())
     }
+    // Extracts the lookup dependant cells and writes assertions of an uninterpreted function using an SMT printer.
+    // Also extract the list of all lookup dependant cells and their corresponding lookup mappings. These data will be used for checking if a under-constrained circuit is false positive.
+    #[cfg(any(
+        feature = "use_zcash_halo2_proofs",
+        feature = "use_pse_halo2_proofs",
+        feature = "use_axiom_halo2_proofs",
+        feature = "use_pse_v1_halo2_proofs",
+    ))]
+    fn decompose_lookups_as_uniterpreted(
+        &mut self,
+        printer: &mut smt::Printer<File>,
+    ) -> Result<(), anyhow::Error> {
+        // As we are defining one uninterpreted function for all lookups, we only need to define it once.
```

Comment:
I think this needs discussion -- not all lookups look at the same inputs, right? So I don't think this is the ideal approach. Later, if you try to include equality for these functions, they may actually be different lookups (with different ranges).

Reply:
Fixed this based on our discussion. Now, for each lookup we will have a separate uninterpreted function for each lookup.